xbmc/pvr/epg/Epg.cpp

```diff
@@ -199,6 +199,29 @@ CPVREpgInfoTagPtr CPVREpg::GetTagNext() const
   return CPVREpgInfoTagPtr();
 }
 
+CPVREpgInfoTagPtr CPVREpg::GetTagPrevious() const
+{
+  CPVREpgInfoTagPtr nowTag(GetTagNow());
+  if (nowTag)
+  {
+    CSingleLock lock(m_critSection);
+    std::map<CDateTime, CPVREpgInfoTagPtr>::const_iterator it = m_tags.find(nowTag->StartAsUTC());
+    if (it != m_tags.end() && --it != m_tags.end())
+      return it->second;
+  }
+  else if (Size() > 0)
+  {
+    /* return the first event that is in the past */
```

Comment:
Why can't this code path be used all the time?

Reply:
I mainly borrowed the implementation from `CPVREpg::GetTagNext()`. I guess it's for performance reasons? First code path is faster, because no sequential seek involved.